packages/vite/src/node/utils.ts

```diff
@@ -865,6 +866,16 @@ export async function resolveHostname(
     // If passed --host in the CLI without arguments
     host = undefined // undefined typically means 0.0.0.0 or :: (listen on all IPs)
   } else {
+    // check if the host is valid
+    if (
+      optionsHost !== 'localhost' &&
+      !wildcardHosts.has(optionsHost) &&
+      !net.isIP(optionsHost)
```

Comment:
I guess the problem here is that `optionsHost` has a non-string value (number) when a number is passed to `--host`. Maybe there's a option to disable that conversion in cac? If not, we can reconvert it to string after getting the value from cac.

Reply:
`cli.option` seems to support a third parameter setting type, I'll give it a try.

Reply:
I guess it's impossible to check whether it's a domain or not.
According to Wikipedia, hostnames can be started with digits, so I guess `0` is a valid hostname.
https://en.wikipedia.org/wiki/Hostname#:~:text=The%20Internet%20standards,digits.%5B5%5D
Adding `127.0.0.1 0` to `C:\Windows\System32\drivers\etc\hosts` and running `npx sirv-cli --host 0` and then executing `curl http://0:8080` worked on my Windows machine.
I don't think people should do this though.

Reply:
The situation you present is so unique that I'm completely unaware of this practice. Exactly checking domains is so difficult, maybe we should just keep checking IP?

I tried to set the host option type parameter `{type: [String]}`, but it seems that it cannot be set directly, otherwise the Boolean type will become a string `'true'` and the test will fail.

I think the current check logic seems to handle this situation, maybe it doesn't need to be converted to a string?

Reply:
Added type parameter.

Reply:
> Exactly checking domains is so difficult, maybe we should just keep checking IP?

I think we should at least accept `^[a-z][\w.]*$` (e.g. `foo.com`, `local`).
If the type parameter works, I think we can remove the `net.isIP`.

Reply:
Maybe this whole if statement can be removed, and we don't need to check the domain?

Reply:
Ah, yes, I meant that.